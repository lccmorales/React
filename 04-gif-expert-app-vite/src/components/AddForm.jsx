import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddForm = ( { setCategories, setLimitNumGifs } ) => {
    const [inputValueSearch, setInputValueSearch] = useState('');
    const [inputValueLimit, setInputValueLimit] = useState('');

    const handleInputSearchChange = (e) => {
        setInputValueSearch(e.target.value);
        console.log('handleInputSearchChange - llamado');
    }

    const handleInputLimitChange = (e) => {
        setInputValueLimit(e.target.value);
        console.log('handleInputLimitChange - llamado');
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit - llamado', inputValueSearch);
        if (inputValueSearch.trim().length > 3 && parseInt(inputValueLimit.trim()) > 0) {
            setCategories( cats => [ inputValueSearch, ...cats,] );
            setInputValueSearch('');
            setLimitNumGifs( parseInt(inputValueLimit) );
            setInputValueLimit('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValueSearch }</p>
            <p>{ inputValueLimit }</p>
            <h3>Busqueda</h3>
            <input
                className='search'
                type="text"
                value={ inputValueSearch }
                onChange={ handleInputSearchChange }
            />

            <h3>Cantidad de Gif's a mostrar</h3>
            <input
                type="number"
                value={ inputValueLimit }
                onChange={ handleInputLimitChange }
            />

            <button type="submit">Buscar</button>
        </form>
    )
}

AddForm.propTypes = {
    setCategories: PropTypes.func.isRequired,
    setLimitNumGifs: PropTypes.func.isRequired
};