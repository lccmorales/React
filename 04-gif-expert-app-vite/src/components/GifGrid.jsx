import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ( { category, limitNumGifs } ) => {
    const { images, loading } = useFetchGifs( category, limitNumGifs );

    return (
        <>
        <h3 className='animate__animated animate__flash' aria-label='titulo'>{ category }</h3>

        { loading && <p className='animate__animated animate__flash'>Cargando...</p> }
        
        <div className='card-grid'>
            {
                images.map( img => (
                    <GifGridItem
                        key={ img.id }
                        { ...img }
                    />
                ))
            }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limitNumGifs: PropTypes.number.isRequired
}