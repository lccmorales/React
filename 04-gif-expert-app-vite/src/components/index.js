export * from './AddForm';
export * from './GifGrid';
export * from './GifGridItem';