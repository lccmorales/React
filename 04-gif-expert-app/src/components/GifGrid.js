import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types';

export const GifGrid = ( {category} ) => {

    const [count, setCount] = useState(0);

    useEffect( () => {
        getGifs();
    }, []);

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Arcane&limit=10&api_key=wGFpRizoH8FkJI4Dbmue2wSoctrFUqGq';
        const resp = await fetch ( url );
        const { data } = await resp.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    return (
        <>
            <h3>{ category }</h3>
            <h1>{ count }</h1>
            <button onClick = { () => setCount(count + 1) }>+1</button>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}