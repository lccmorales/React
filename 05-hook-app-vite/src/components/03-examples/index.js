export * from './Button';
export * from './LoadingQuote';
export * from './MultipleCustomHooks';
export * from './Quote';