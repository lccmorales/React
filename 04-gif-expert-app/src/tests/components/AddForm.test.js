import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddForm } from '../../components/AddForm';

describe('Pruebas en <AddForm />', () => {
    const setCategories = jest.fn();
    const setLimitNumGifs = jest.fn();
    let wrapper = shallow( <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs } />);

    beforeEach(() => {
        // jest.clearAllMocks();
        // wrapper = shallow( <AddForm setCategories={ setCategories } setLimitNumGifs={ setLimitNumGifs } />);
    });

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de cambiar la caja de texto de Categorias', () => {
        const inputCategorias = wrapper.find('input').at(0);
        const value = 'Hola Mundo';
        inputCategorias.simulate('change', { target: { value }});
        expect( wrapper.find('p').at(0).text().trim() ).toBe( value );
    })

    test('debe de cambiar la caja de texto de Cantidad de Gifs a mostrar', () => {
        const inputLimit = wrapper.find('input').at(1);
        const value = '4';
        inputLimit.simulate('change', { target: { value: value }});
        expect( wrapper.find('p').at(1).text().trim() ).toBe( value );
    })

    // test('NO debe de postear la informaciÃ³n con submit', () => {
    //     wrapper.find('form').simulate('submit', { preventDefault(){} });
    //     expect( setCategories ).not.toHaveBeenCalled();
    // })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Hola Mundo';
        // 1. simular el inputChange
        wrapper.find('.search').simulate('change', { target: { value }});
        // 2. simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });
        // 3. setCategories se debe de haber llamado
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));
        // 4. el valor del input debe de estar ''
        expect( wrapper.find('input').at(0).prop('value') ).toBe('');
    })
})