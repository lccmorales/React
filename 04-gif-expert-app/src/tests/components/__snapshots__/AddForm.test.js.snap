// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddForm /> debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <p />
  <h3>
    Busqueda
  </h3>
  <input
    className="search"
    onChange={[Function]}
    type="text"
    value=""
  />
  <h3>
    Cantidad de Gif's a mostrar
  </h3>
  <input
    onChange={[Function]}
    type="number"
    value=""
  />
  <button
    type="submit"
  >
    Buscar
  </button>
</form>
`;
